### PR TITLE
feat: add github ci

### PR DESCRIPTION
**Pull request by Gwemox**
Here is a functional CI with Github Actions. It has helped me identify several issues:

- The PHP minimum version is 8.1 in the `composer.json`, while the tests were on 7.3 and 7.4. And the documentation still displays 7.3 as the minimum.
- We were trying to test a JSON CAST for MariaDB when it is not available.

### CODE REVIEW

File: tests/Oro/Tests/ORM/AST/Query/Functions/fixtures/mysql/cast.yml [hunk 1]
```diff
@@ -81,13 +81,6 @@
   expectedResult:
       - '2014-01-04 05:06:07'
 
-- functions:
```
Comment (x86demon) [inline tests/Oro/Tests/ORM/AST/Query/Functions/fixtures/mysql/cast.yml:84]: Should be supported
https://dev.mysql.com/doc/refman/8.0/en/json.html

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+name: CI
```
Comment (x86demon) [inline .github/workflows/ci.yml:1]: :+1:

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php: [8.1]
+        db_image_name: [mysql, mariadb, percona, postgres]
+        include:
+          - db_image_name: mysql
+            db_image_version: 8
+            db: mysql
+            vars: '-e MYSQL_PASSWORD=travis -e MYSQL_DATABASE=doctrine_extensions_tests -e MYSQL_USER=travis -e MYSQL_ROOT_PASSWORD=travis'
```
Comment (x86demon) [inline .github/workflows/ci.yml:16]: Please check the DB credentials, it seems that the ones used are actual for travis CI.

**Review by x86demon — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (Gwemox) [reply .github/workflows/ci.yml:16]: I don't understand what you expect from me.

Do you want me to switch to another User / Password?
Is Travis CI still used anywhere?

File: tests/Oro/Tests/ORM/AST/Query/Functions/fixtures/mysql/cast.yml (hunk 1 above)
Comment (Gwemox) [reply tests/Oro/Tests/ORM/AST/Query/Functions/fixtures/mysql/cast.yml]: Not in MariaDB https://mariadb.com/kb/en/cast/

I reverted the commit and the tests fail

File: .github/workflows/ci.yml (hunk 3 above)
Comment (x86demon) [reply .github/workflows/ci.yml:16]: Travis CI is no longer used.

 > Do you want me to switch to another User / Password?

Probably yes, the mention of the Travis looks a bit strange here.

File: tests/Oro/Tests/ORM/AST/Query/Functions/fixtures/mysql/cast.yml (hunk 1 above)
Comment (x86demon) [reply tests/Oro/Tests/ORM/AST/Query/Functions/fixtures/mysql/cast.yml]: MariaDB has never been supported by oro/doctrine-extensions. 
Support for the JSON type is critical and cannot be dropped.

**Review by x86demon — CHANGES_REQUESTED** [review]

File: tests/Oro/Tests/ORM/AST/Query/Functions/fixtures/mysql/cast.yml (hunk 1 above)
Comment (Gwemox) [reply tests/Oro/Tests/ORM/AST/Query/Functions/fixtures/mysql/cast.yml]: Previous CI (Travis) have tests using MariaDB.

**Review by x86demon — APPROVED** [review]

**Comment by x86demon** [comment]
@Gwemox Thank you very much for your contribution. Great job :1st_place_medal:

**Pull request merged**